src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedParameterSymbol.cs

```diff
@@ -36,6 +37,11 @@ internal abstract class SynthesizedParameterSymbolBase : ParameterSymbol
             _ordinal = ordinal;
             _refKind = refKind;
             _name = name;
+
+            if (this.RefKind == RefKind.RefReadOnly)
+            {
+                this.DeclaringCompilation.EnsureIsReadOnlyAttributeExists(this);
```

Comment:
Also, it doesn't look like we are going to get good error location from this symbol.

---
In reply to: [113566738](https://github.com/dotnet/roslyn/pull/18715#discussion_r113566738) [](ancestors = 113566738,113566464)